core/src/main/java/com/exadel/etoolbox/linkinspector/core/servlets/ReplaceByPatternServlet.java

```diff
@@ -194,53 +191,74 @@ private List<UpdatedItem> processResources(Collection<GridResource> gridResource
             LOG.warn("Replacement failed, session is null. Pattern: {}, replacement: {}", linkPattern, replacement);
             return Collections.emptyList();
         }
-        List<GridResource> filteredGridResources = filterGridResources(gridResources, linkPattern, session.get());
+        List<List<GridResource>> filteredGridResources = filterGridResources(gridResources, linkPattern, session.get());
         if (isBackup && !isDeactivated) {
-            createBackupPackage(filteredGridResources, session.get());
+            createBackupPackage(filteredGridResources.stream().flatMap(Collection::stream).collect(Collectors.toList()), session.get());
         }
         return replaceByPattern(filteredGridResources, isDryRun, linkPattern, replacement, resourceResolver);
     }
 
-    private List<GridResource> filterGridResources(Collection<GridResource> gridResources,
-                                                   String linkPattern,
-                                                   Session session) {
+    private List<List<GridResource>> filterGridResources(Collection<List<GridResource>> gridResources,
+                                                         String linkPattern,
+                                                         Session session) {
         Pattern pattern = Pattern.compile(linkPattern);
-        return gridResources.stream()
-                .filter(gridResource ->
-                        StringUtils.isNoneBlank(gridResource.getHref(), gridResource.getResourcePath(), gridResource.getPropertyName())
-                )
-                .filter(gridResource -> pattern.matcher(gridResource.getHref()).find())
-                .filter(gridResource ->
-                        repositoryHelper.hasReadWritePermissions(session, gridResource.getResourcePath()))
-                .limit(maxUpdatedItemsCount)
-                .collect(Collectors.toList());
+        return gridResources.stream().map(resources ->
+                resources.stream()
+                        .filter(gridResource ->
+                                StringUtils.isNoneBlank(gridResource.getHref(), gridResource.getResourcePath(), gridResource.getPropertyName())
+                        )
+                        .filter(gridResource -> pattern.matcher(gridResource.getHref()).find())
+                        .filter(gridResource ->
+                                repositoryHelper.hasReadWritePermissions(session, gridResource.getResourcePath()))
+                        .limit(maxUpdatedItemsCount)
```

Comment:
What is the logic behind this parameter? Looks like I -- the author -- ask to update all my links, but only N links are updated, and I have no idea why